insights/client/crypto.py

```diff
@@ -0,0 +1,205 @@
+import errno
+import os.path
+import logging
+import shutil
+import tempfile
+import subprocess
+
+logger = logging.getLogger(__file__)
+
+
+class CommandResult(object):
+    """Output of a command.
+
+    Attributes:
+        ok (bool): Result of an operation.
+        return_code (int): Return code of the operation.
+        stdout (str): Standard output of the command.
+        stderr (str): Standard error of the command.
+        _command: An optional reference to the object that created the result
+    """
+    def __init__(self, ok, return_code, stdout, stderr):
+        self.ok = ok
+        self.return_code = return_code
+        self.stdout = stdout
+        self.stderr = stderr
+        self._command = None
+
+    def __str__(self):
+        return (
+            "<{cls} ok={ok} return_code={code} stdout={out} stderr={err}>"
+        ).format(
+            cls=self.__class__.__name__,
+            ok=self.ok, code=self.return_code,
+            out=self.stdout, err=self.stderr,
+        )
```

Comment:
This looks like a [dataclass](https://docs.python.org/3/library/dataclasses.html). Dataclasses were introduced in Python 3.7 and Insights Client needs to support Python 2.6, but there are _namedtuples_ that do exactly what this class implements.

```suggestion
from collections import namedtuple
CommandResult = namedtuple("CommandResult", ["ok", "return_code", "stdout", "stderr", "command"])
```

I still doubt, however, that this abstraction is actually needed. Although the structure looks nice and clean, it increases complexity. It may be enough to just work with native Python shell command handling.

Moreover, it looks pretty low-level for something that should live in this specifict _crypto_ module. There are other shell command calls in the codebase that may make use of this structure, if we wanted to keep it.

Reply:
I wouldn't call this an increased complexity. It just wraps the data, instead of passing them around in an anonymous dictionary.
I've renamed it to GPGCommandResult to make its existence more clear.

Reply:
_GPGCommandResult_ is definitely better than just plain _CommandResult_. Crypto ≠ GPG, so the module name is not enough and the GPG prefix is great. 👍🏻

I am not sure about the capitalization: whether to use _GPGCommandResult_ or _GpgCommandResult_. I researched on the matter and there is no consensus. My personal taste would be _Gpg_, but lot of big projects use _GPG_. Let’s find out how acronyms are capitalized in the existing parts of the Client and stick to that.

This addressed only the naming. I have more thoughts on the actual code. I‘ll post them separately.